### ISSUE TITLE
508-defect-3 [SCREENREADER]: Resources and support search result metadata should be linked to their respective headings using `aria-describedby`

### ISSUE THREAD

**Issue by joshkimux** [department-of-veterans-affairs/va.gov-team#29789]
# [508-defect-3](https://github.com/department-of-veterans-affairs/va.gov-team/blob/master/platform/accessibility/guidance/defect-severity-rubric.md#508-defect-3)

## Feedback framework

- **❗️ Must** for if the feedback must be applied 
- **⚠️ Should** if the feedback is best practice 
- **✔️ Consider** for suggestions/enhancements 

## Definition of done

1. Review and acknowledge feedback.
1. Fix and/or document decisions made.
1. Accessibility specialist will close ticket after reviewing documented decisions / validating fix.

## Point of Contact

**VFS Point of Contact:** Josh

## Details

Screen reader users who navigate search results using headings shortcuts or the `tab` key will likely miss metadata descriptions that are placed above each link heading search result.

https://user-images.githubusercontent.com/14154792/132714525-3e060837-a364-4801-8db9-15e047f6ee1a.mov

Since screen reader users read a page top to bottom, the metadata tags (e.g. "Multiple FAQs") will likely be undetected if they use the `tab` key since it's unnatural to read _backwards_. For example, someone might tab to the first result "Veteran Identification Card" and then read the line below it-- but they likely won't tab to the first result and read the line _above_ it.

## Acceptance Criteria

- [ ] Metadata is announced to screen reader users using `aria-describedby`

## Steps to Recreate

1. Enter `https://www.va.gov/resources/search/?query=veteran` in your browser
2. Turn on your screen reader and tab to the first search result
3. Confirm the metadata above the search result is not read out

## Proposed Solution 

[Find a Form used `aria-describedby` to design a more inclusive experience for this same issue.](https://github.com/department-of-veterans-affairs/va.gov-team/issues/26353#issuecomment-892829129). I recommend doing the same here.

### Before

```html
<div>
  <dfn class="sr-only">Article type: </dfn>
    Multiple FAQs
</div>
<h2>
  <a href="/resources/veteran-identification-card-vic-faqs">Veteran Identification Card (VIC) FAQs</a>
</h2>
<p class="vads-u-margin-bottom--0">You may already have an identification card that you can use to show you’re a Veteran. But if you don’t, you can apply for a VIC to get discounts offered to Veterans.</p>
```

### After

```html
<div id="unique-id"> <!-- Add a unique id to the div that contains the article type -->
  <dfn class="sr-only">Article type: </dfn>
    Multiple FAQs
</div>
<h2 aria-describedby="unique-id"> <!-- Use aria-describedby to reference the unique id-->
  <a href="/resources/veteran-identification-card-vic-faqs">Veteran Identification Card (VIC) FAQs</a>
</h2>
<p class="vads-u-margin-bottom--0">You may already have an identification card that you can use to show you’re a Veteran. But if you don’t, you can apply for a VIC to get discounts offered to Veterans.</p>
```
`aria-describedby` will then automatically append the description (article type) to the search result:

<img width="1489" alt="Screen Shot 2021-09-09 at 11 31 34 AM" src="https://user-images.githubusercontent.com/14154792/132716382-c5fa6814-a858-455c-93a5-0c5f70c3089d.png">

**Comment by joshkimux** [comment]
Tagging @brianalloyd for awareness! 😄
